app/src/main/java/com/example/henzoshimada/feeltrip/EditMoodActivity.java

```diff
@@ -161,4 +225,32 @@ private void addItemsOnSocialSitualtionSpinner(){
         socialSituationSpinner.setAdapter(socialSituationAdapter);
     }
 
+    // get the selected dropdown list value
+    public void addListenerOnSubmitButton() {
+
+        emotionalStateSpinner = (Spinner) findViewById(R.id.emotional_state_spinner);
+        socialSituationSpinner = (Spinner) findViewById(R.id.social_event_spinner);
+        Button submitButton = (Button) findViewById(R.id.post_mood_button);
+
+        submitButton.setOnClickListener(new View.OnClickListener() {
+
+            @Override
+            public void onClick(View v) {
+
+                emotionalState = String.valueOf(emotionalStateSpinner.getSelectedItem());
+                socialSit = String.valueOf(socialSituationSpinner.getSelectedItem());
+
+                try {
+                    submitMood();
+                    Toast.makeText(getApplicationContext(), "Successfully posted!", Toast.LENGTH_SHORT).show();
+                    finish();
+                } catch (DescriptionTooLongException e) {
+                    Toast.makeText(getApplicationContext(), "Your description is too long.", Toast.LENGTH_SHORT).show();
```

Comment:
Can we expand our exception criteria? Seems a bit silly to limit the user to 3 words/20 chars...